File: Micro_core/API/DataModels/reports/EntityType.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using API.Interfaces;

namespace API.DataModels.reports
{
    [Table("tbl_Entity_Type")]
    public class EntityType: IBaseModel
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }

        [DefaultValue(false)]
        public bool Active { get; set; }
        [DefaultValue(false)]
        public bool Delete { get; set; }

        public ICollection<EntityPages> Entitypages { get; } = new List<EntityPages>();
        public ICollection<Fields> Fields { get; } = new List<Fields>(); 
    }
}

File: Micro_core/ClientApp/app/models/entityType.ts
import { Pages } from './pages';
import { Fields } from './fields';
export interface EntityType {
    id: number;
    name: string;
    active: boolean;
    delete: boolean;
    pages: Pages[];
    fields: Fields[];
}
export interface EntityTypeFunc extends Function {
    prototype: EntityType;
    new (): EntityType;
}
var EntityType: EntityTypeFunc;